### PR TITLE
Enable Throttling when transcoding without Hardware-Acceleration

### PR DESCRIPTION
**Pull request by geilername**
**Changes**
Check if Hardware-Acceleration is enabled when playback is starting. If it's enabled we disable throttling else we enable it.

**Question**
If throttling Hardware-Accelerated transcodes is OK I'd change this again.

### CODE REVIEW

File: MediaBrowser.Api/Playback/BaseStreamingService.cs [hunk 1]
```diff
@@ -327,6 +327,18 @@ private void StartThrottler(StreamState state, TranscodingJob transcodingJob)
...
         private bool EnableThrottling(StreamState state)
         {
+            var encodingOptions = ServerConfigurationManager.GetEncodingOptions();
+
+            // enable throttling when not using hardware acceleration
+            if (encodingOptions.HardwareAccelerationType == string.Empty)
+            {
+                return state.InputProtocol == MediaProtocol.File &&
+                       state.RunTimeTicks.HasValue &&
+                       state.RunTimeTicks.Value >= TimeSpan.FromMinutes(5).Ticks &&
```
Comment (JustAMan) [inline MediaBrowser.Api/Playback/BaseStreamingService.cs]: this feels somewhat wrong, as I thought that `RunTimeTicks` is for play position, not transcode... was I wrong?

Comment (geilername) [reply MediaBrowser.Api/Playback/BaseStreamingService.cs]: i just used what was there, tried it with some files and it looked like it worked

Comment (JustAMan) [reply MediaBrowser.Api/Playback/BaseStreamingService.cs]: Well, according to my search I was probably wrong. `state.RunTimeTicks` seems to represent total length of the item, so this check effectively says "enable throttling on items longer than 5 minutes".

I still don't like current implementation of the throttler, but we can give it a chance.

Comment (geilername) [reply MediaBrowser.Api/Playback/BaseStreamingService.cs]: well thats an ok reason to disable throttling i think? a 5min file should be transcoded pretty much instantly and while paused the ffmpeg process is using RAM

**Comment by Bond-009** [comment]
Why don't we remove the commented out code?

**Comment by JustAMan** [comment]
@Bond-009 
> Why don't we remove the commented out code?

I've removed that code, can you post your review?

**Comment by Artiume** [comment]
hm... i think im getting issues with the transcoder throttling.

im having a hevc file pause, the logs have this
```
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47973.ts' for writing
frame= 5339 fps= 89 q=25.0 size=N/A time=00:03:43.48 bitrate=N/A speed=3.72x    
Enter command: <target>|all <time>|-1 <command>[ <argument>]
```

digging into wtf that's about, it pulls up an emby post for throttling, still looking into whats going on https://emby.media/community/index.php?/topic/18703-one-small-step-for-ffmpeg-one-giant-leap-for-emby-looking-for-c-developers-to-help-with-transcode-throttling/

**Comment by Artiume** [comment]
LOC 358   string.Equals(GetVideoEncoder(state), "libx264", StringComparison.OrdinalIgnoreCase);

this is missing from the new merge, perhaps thats whats missing? I tried re-adding it but GetVideoEncoder is no longer available. I reverted the PR on my master and no longer have playback issues. Video in question is HEVC 10Bit. @geilername did you test these file types in your PR?

**Comment by geilername** [comment]
> LOC 358 string.Equals(GetVideoEncoder(state), "libx264", StringComparison.OrdinalIgnoreCase);
> 
> this is missing from the new merge, perhaps thats whats missing? I tried re-adding it but GetVideoEncoder is no longer available. I reverted the PR on my master and no longer have playback issues. Video in question is HEVC 10Bit. @geilername did you test these file types in your PR?

well I tested with x265 but only with non HDR content

**Comment by Artiume** [comment]
10bit doesn't necessarily mean hdr. I don't have any hdr content 😬

**Comment by Artiume** [comment]
Here's the logs, let me know if you can't resolve it and i'll make a proper ticket for it. 

```
{"Protocol":"File","Id":"da581b2127e45ab06740e6282c6b8758","Path":"/media/anime/Slayers/Season 3/Slayers - S03E03 - Where'd That Arrogant Guy Go HDTV-1080p.mkv","Type":"Default","Container":"mkv,webm","Size":683591296,"Name":"Slayers - S03E03 - Where'd That Arrogant Guy Go HDTV-1080p","IsRemote":false,"ETag":"2c12d9d2c2503e7472a2205075351f82","RunTimeTicks":14519249920,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"RequiresClosing":false,"RequiresLooping":false,"SupportsProbing":true,"VideoType":"VideoFile","MediaStreams":[{"Codec":"aac","Language":"eng","TimeBase":"1/1000","CodecTimeBase":"1/48000","DisplayTitle":"Eng AAC stereo Default","IsInterlaced":false,"ChannelLayout":"stereo","BitRate":192000,"Channels":2,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Profile":"Main","Type":"Audio","Index":0,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Level":0},{"Codec":"hevc","TimeBase":"1/1000","CodecTimeBase":"1001/24000","VideoRange":"SDR","DisplayTitle":"1080P HEVC","IsInterlaced":false,"BitRate":3766537,"RefFrames":1,"IsDefault":true,"IsForced":false,"Height":1080,"Width":1920,"AverageFrameRate":23.976025,"RealFrameRate":23.976025,"Profile":"Main 10","Type":"Video","AspectRatio":"16:9","Index":1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"PixelFormat":"yuv420p10le","Level":186},{"Codec":"aac","Language":"jpn","TimeBase":"1/1000","CodecTimeBase":"1/48000","DisplayTitle":"Jpn AAC stereo","IsInterlaced":false,"ChannelLayout":"stereo","BitRate":192000,"Channels":2,"SampleRate":48000,"IsDefault":false,"IsForced":false,"Profile":"LC","Type":"Audio","Index":2,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Level":0},{"Codec":"ass","Language":"eng","TimeBase":"1/1000","CodecTimeBase":"0/1","localizedUndefined":"Undefined","localizedDefault":"Default","localizedForced":"Forced","DisplayTitle":"Eng","IsInterlaced":false,"IsDefault":false,"IsForced":false,"Type":"Subtitle","Index":3,"IsExternal":false,"IsTextSubtitleStream":true,"SupportsExternalStream":true,"Level":0}],"MediaAttachments":[],"Formats":[],"Bitrate":4150537,"RequiredHttpHeaders":{}}

/usr/local/bin/ffmpeg -f matroska,webm -i file:"/media/anime/Slayers/Season 3/Slayers - S03E03 - Where'd That Arrogant Guy Go HDTV-1080p.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:1 -map 0:0 -map -0:s -codec:v:0 libx264 -pix_fmt yuv420p -preset veryfast -crf 23 -maxrate 7533074 -bufsize 15066148 -profile:v high -level 4.1 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none  -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "scale=trunc(min(max(iw\,ih*dar)\,1920)/2)*2:trunc(ow/dar/2)*2" -copyts -vsync -1 -codec:a:0 copy -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de479%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de479.m3u8"


Input #0, matroska,webm, from 'file:/media/anime/Slayers/Season 3/Slayers - S03E03 - Where'd That Arrogant Guy Go HDTV-1080p.mkv':
  Metadata:
    encoder         : libebml v1.3.6 + libmatroska v1.4.9
    creation_time   : 2019-05-17T07:17:14.000000Z
  Duration: 00:24:11.93, start: 0.000000, bitrate: 3766 kb/s
    Stream #0:0(eng): Audio: aac (Main), 48000 Hz, stereo, fltp (default)
    Metadata:
      BPS-eng         : 142365
      DURATION-eng    : 00:24:11.925000000
      NUMBER_OF_FRAMES-eng: 68059
      NUMBER_OF_BYTES-eng: 25837991
      _STATISTICS_WRITING_APP-eng: mkvmerge v31.0.0 ('Dolores In A Shoestand') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-05-17 07:17:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:1: Video: hevc (Main 10), yuv420p10le(tv), 1920x1080, SAR 1:1 DAR 16:9, 23.98 fps, 23.98 tbr, 1k tbn, 23.98 tbc (default)
    Metadata:
      BPS-eng         : 3727657
      DURATION-eng    : 00:22:35.354000000
      NUMBER_OF_FRAMES-eng: 32496
      NUMBER_OF_BYTES-eng: 631536934
      _STATISTICS_WRITING_APP-eng: mkvmerge v31.0.0 ('Dolores In A Shoestand') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-05-17 07:17:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:2(jpn): Audio: aac (LC), 48000 Hz, stereo, fltp
    Metadata:
      BPS-eng         : 151139
      DURATION-eng    : 00:22:34.902000000
      NUMBER_OF_FRAMES-eng: 63511
      NUMBER_OF_BYTES-eng: 25597474
      _STATISTICS_WRITING_APP-eng: mkvmerge v31.0.0 ('Dolores In A Shoestand') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-05-17 07:17:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:3(eng): Subtitle: ass
    Metadata:
      BPS-eng         : 121
      DURATION-eng    : 00:20:58.420000000
      NUMBER_OF_FRAMES-eng: 267
      NUMBER_OF_BYTES-eng: 19187
      _STATISTICS_WRITING_APP-eng: mkvmerge v31.0.0 ('Dolores In A Shoestand') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-05-17 07:17:14
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:4: Attachment: ttf
    Metadata:
      filename        : BAARS___.TTF
      mimetype        : application/x-truetype-font
Stream mapping:
  Stream #0:1 -> #0:0 (hevc (native) -> h264 (libx264))
  Stream #0:0 -> #0:1 (copy)
Press [q] to stop, [?] for help
[libx264 @ 0x5598b68ef1c0] using SAR=1/1
[libx264 @ 0x5598b68ef1c0] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[libx264 @ 0x5598b68ef1c0] profile Main, level 4.1
[libx264 @ 0x5598b68ef1c0] 264 - core 155 - H.264/MPEG-4 AVC codec - Copyleft 2003-2018 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x1:0 me=dia subme=0 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=4 chroma_me=0 trellis=0 8x8dct=0 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=34 lookahead_threads=5 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=250 keyint_min=23 scenecut=40 intra_refresh=0 rc_lookahead=10 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=7533 vbv_bufsize=15066 crf_max=0.0 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de4790.ts' for writing
Output #0, hls, to '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de479.m3u8':
  Metadata:
    encoder         : Lavf58.29.100
    Stream #0:0: Video: h264 (libx264), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], q=-1--1, 23.98 fps, 90k tbn, 23.98 tbc (default)
    Metadata:
      encoder         : Lavc58.54.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 7533000/0/0 buffer size: 15066000 vbv_delay: -1
    Stream #0:1: Audio: aac (Main), 48000 Hz, stereo, fltp (default)
frame=   10 fps=0.0 q=0.0 size=N/A time=00:00:01.17 bitrate=N/A speed=2.34x    
frame=   10 fps=0.0 q=0.0 size=N/A time=00:00:01.17 bitrate=N/A speed=2.34x    
frame=   12 fps= 11 q=0.0 size=N/A time=00:00:01.34 bitrate=N/A speed=1.22x    
frame=   14 fps=7.4 q=0.0 size=N/A time=00:00:01.34 bitrate=N/A speed=0.711x    
frame=   15 fps=6.0 q=0.0 size=N/A time=00:00:01.36 bitrate=N/A speed=0.547x    
frame=   18 fps=5.3 q=0.0 size=N/A time=00:00:01.51 bitrate=N/A speed=0.448x    
frame=   20 fps=4.6 q=0.0 size=N/A time=00:00:01.68 bitrate=N/A speed=0.387x    
frame=   21 fps=4.2 q=0.0 size=N/A time=00:00:01.70 bitrate=N/A speed=0.341x    
frame=   26 fps=4.6 q=0.0 size=N/A time=00:00:01.85 bitrate=N/A speed=0.331x    
frame=   28 fps=4.5 q=0.0 size=N/A time=00:00:01.85 bitrate=N/A speed=0.298x    
frame=   29 fps=4.3 q=0.0 size=N/A time=00:00:01.87 bitrate=N/A speed=0.275x    
frame=   33 fps=4.4 q=0.0 size=N/A time=00:00:02.02 bitrate=N/A speed=0.272x    
frame=   35 fps=4.4 q=0.0 size=N/A time=00:00:02.19 bitrate=N/A speed=0.274x    
frame=   37 fps=4.3 q=0.0 size=N/A time=00:00:02.19 bitrate=N/A speed=0.256x    
frame=   41 fps=4.4 q=0.0 size=N/A time=00:00:02.53 bitrate=N/A speed=0.274x    
frame=   42 fps=4.3 q=0.0 size=N/A time=00:00:02.56 bitrate=N/A speed=0.261x    
frame=   45 fps=4.4 q=0.0 size=N/A time=00:00:02.70 bitrate=N/A speed=0.262x    
frame=   50 fps=4.6 q=28.0 size=N/A time=00:00:02.87 bitrate=N/A speed=0.263x    
frame=   53 fps=4.6 q=28.0 size=N/A time=00:00:03.04 bitrate=N/A speed=0.263x    
frame=   80 fps=6.6 q=28.0 size=N/A time=00:00:04.07 bitrate=N/A speed=0.336x    
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de4791.ts' for writing

truncated logs

[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47970.ts' for writing
frame= 5104 fps= 88 q=28.0 size=N/A time=00:03:33.71 bitrate=N/A speed=3.68x    
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47971.ts' for writing
frame= 5166 fps= 88 q=28.0 size=N/A time=00:03:36.21 bitrate=N/A speed=3.69x    
frame= 5223 fps= 88 q=28.0 size=N/A time=00:03:38.55 bitrate=N/A speed= 3.7x    
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47972.ts' for writing
frame= 5280 fps= 89 q=28.0 size=N/A time=00:03:40.88 bitrate=N/A speed=3.71x    
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47973.ts' for writing
frame= 5339 fps= 89 q=25.0 size=N/A time=00:03:43.48 bitrate=N/A speed=3.72x    
Enter command: <target>|all <time>|-1 <command>[ <argument>]

Parse error, at least 3 arguments were expected, only -1 given in string ''
[hls @ 0x5598b68ed740] Opening '/ram_transcode/3ad7ab7ce6172d6db6fbc6f5c77de47974.ts' for writing
frame= 5356 fps=9.3 q=-1.0 Lsize=N/A time=00:03:44.25 bitrate=N/A speed=0.39x    
video:108118kB audio:3897kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
[libx264 @ 0x5598b68ef1c0] frame I:127   Avg QP:16.99  size: 83559
[libx264 @ 0x5598b68ef1c0] frame P:1855  Avg QP:23.35  size: 38092
[libx264 @ 0x5598b68ef1c0] frame B:3374  Avg QP:26.32  size:  8726
[libx264 @ 0x5598b68ef1c0] consecutive B-frames: 10.3% 13.7% 10.1% 65.9%
[libx264 @ 0x5598b68ef1c0] mb I  I16..4: 58.2%  0.0% 41.8%
[libx264 @ 0x5598b68ef1c0] mb P  I16..4: 30.5%  0.0%  0.0%  P16..4: 28.6%  0.0%  0.0%  0.0%  0.0%    skip:40.9%
[libx264 @ 0x5598b68ef1c0] mb B  I16..4:  3.5%  0.0%  0.0%  B16..8: 16.3%  0.0%  0.0%  direct: 6.2%  skip:74.1%  L0:42.1% L1:46.4% BI:11.6%
[libx264 @ 0x5598b68ef1c0] coded y,uvDC,uvAC intra: 28.6% 52.5% 10.9% inter: 7.8% 9.8% 0.2%
[libx264 @ 0x5598b68ef1c0] i16 v,h,dc,p: 42% 25% 15% 18%
[libx264 @ 0x5598b68ef1c0] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 27% 22% 14%  9%  6%  5%  5%  5%  6%
[libx264 @ 0x5598b68ef1c0] i8c dc,h,v,p: 39% 23% 24% 14%
[libx264 @ 0x5598b68ef1c0] Weighted P-Frames: Y:4.0% UV:2.4%
[libx264 @ 0x5598b68ef1c0] kb/s:3964.82
```

**Comment by geilername** [comment]
Sorry for the late response, I've been very busy om the last weeks. I tried it with hevc files but was not able to reproduce your error, I'm using the official jellyfin/jellyfin docker image maybe you are using some other version ffmpeg?

`General
Unique ID                                : 122837973198687415740616723285952862096 (0x5C69BBABBA7EA7215FEFB719215E9790)
Complete name                            : <filepath>
Format                                   : Matroska
Format version                           : Version 4
File size                                : 2.50 GiB
Duration                                 : 1 h 32 min
Overall bit rate                         : 3 882 kb/s
Encoded date                             : UTC 2020-01-26 13:09:16
Writing application                      : mkvmerge v42.0.0 ('Overtime') 64-bit
Writing library                          : libebml v1.3.10 + libmatroska v1.5.2 / Lavf58.29.100

Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main@L4@Main
Codec ID                                 : V_MPEGH/ISO/HEVC
Duration                                 : 1 h 32 min
Bit rate                                 : 3 112 kb/s
Width                                    : 1 920 pixels
Height                                   : 800 pixels
Display aspect ratio                     : 2.40:1
Frame rate mode                          : Constant
Frame rate                               : 23.976 (24000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Bits/(Pixel*Frame)                       : 0.084
Stream size                              : 2.01 GiB (80%)
Writing library                          : x265 3.2.1+1-b5c86a64bbbe:[Windows][GCC 8.3.0][64 bit] 8bit+10bit+12bit
Encoding settings                        : cpuid=1111039 / frame-threads=3 / numa-pools=8 / wpp / no-pmode / no-pme / no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=1920x800 / interlace=0 / total-frames=0 / level-idc=0 / high-tier=1 / uhd-bd=0 / ref=3 / no-allow-non-conformance / no-repeat-headers / annexb / no-aud / no-hrd / info / hash=0 / no-temporal-layers / open-gop / min-keyint=24 / keyint=240 / gop-lookahead=0 / bframes=4 / b-adapt=2 / b-pyramid / bframe-bias=0 / rc-lookahead=20 / lookahead-slices=5 / scenecut=40 / radl=0 / no-splice / no-intra-refresh / ctu=64 / min-cu-size=8 / no-rect / no-amp / max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=0 / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 / no-constrained-intra / no-strong-intra-smoothing / max-merge=3 / limit-refs=1 / no-limit-modes / me=1 / subme=2 / merange=57 / temporal-mvp / no-hme / weightp / no-weightb / no-analyze-src-pics / deblock=0:0 / no-sao / no-sao-non-deblock / rd=3 / selective-sao=0 / early-skip / rskip / no-fast-intra / no-tskip-fast / no-cu-lossless / b-intra / no-splitrd-skip / rdpenalty=0 / psy-rd=2.00 / psy-rdoq=0.00 / no-rd-refine / no-lossless / cbqpoffs=0 / crqpoffs=0 / rc=crf / crf=23.0 / qcomp=0.60 / qpstep=4 / stats-write=0 / stats-read=0 / ipratio=1.40 / pbratio=1.30 / aq-mode=1 / aq-strength=1.00 / cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 / qpmin=0 / no-const-vbv / sar=1 / overscan=0 / videoformat=5 / range=0 / colorprim=1 / transfer=1 / colormatrix=1 / chromaloc=0 / display-window=0 / cll=0,0 / min-luma=0 / max-luma=255 / log2-max-poc-lsb=8 / vui-timing-info / vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps / no-multi-pass-opt-rps / scenecut-bias=0.05 / no-opt-cu-delta-qp / no-aq-motion / no-hdr / no-hdr-opt / no-dhdr10-opt / no-idr-recovery-sei / analysis-reuse-level=5 / scale-factor=0 / refine-intra=0 / refine-inter=0 / refine-mv=1 / refine-ctu-distortion=0 / no-limit-sao / ctu-info=0 / no-lowpass-dct / refine-analysis-type=0 / copy-pic=1 / max-ausize-factor=1.0 / no-dynamic-refine / no-single-sei / no-hevc-aq / no-svt / no-field / qp-adaptation-range=1.00
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio #1
ID                                       : 2
Format                                   : AC-3
Format/Info                              : Audio Coding 3
Commercial name                          : Dolby Digital
Codec ID                                 : A_AC3
Duration                                 : 1 h 32 min
Bit rate mode                            : Constant
Bit rate                                 : 384 kb/s
Channel(s)                               : 6 channels
Channel layout                           : L R C LFE Ls Rs
Sampling rate                            : 48.0 kHz
Frame rate                               : 31.250 FPS (1536 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -5 ms
Stream size                              : 253 MiB (10%)
Title                                    : Surround
Language                                 : German
Service kind                             : Complete Main
Default                                  : Yes
Forced                                   : No

Audio #2
ID                                       : 3
Format                                   : AC-3
Format/Info                              : Audio Coding 3
Commercial name                          : Dolby Digital
Codec ID                                 : A_AC3
Duration                                 : 1 h 32 min
Bit rate mode                            : Constant
Bit rate                                 : 384 kb/s
Channel(s)                               : 6 channels
Channel layout                           : L R C LFE Ls Rs
Sampling rate                            : 48.0 kHz
Frame rate                               : 31.250 FPS (1536 SPF)
Compression mode                         : Lossy
Delay relative to video                  : -5 ms
Stream size                              : 253 MiB (10%)
Title                                    : Surround
Language                                 : English
Service kind                             : Complete Main
Default                                  : No
Forced                                   : No

Text #1
ID                                       : 4
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 h 19 min
Bit rate                                 : 33 b/s
Count of elements                        : 506
Stream size                              : 19.6 KiB (0%)
Title                                    : full (ger)
Language                                 : German
Default                                  : No
Forced                                   : No

Text #2
ID                                       : 5
Format                                   : UTF-8
Codec ID                                 : S_TEXT/UTF8
Codec ID/Info                            : UTF-8 Plain Text
Duration                                 : 1 h 19 min
Bit rate                                 : 38 b/s
Count of elements                        : 698
Stream size                              : 22.3 KiB (0%)
Title                                    : full (eng)
Language                                 : English
Default                                  : No
Forced                                   : No`

**Comment by Artiume** [comment]
Are you available on matrix? And your sample file is only 8 bit

**Comment by JustAMan** [comment]
What platform you're running it on? There are chances you get an official image without our ffmpeg (as we don't ship it for ARM if I'm not mistaken), in which case you ended up with a pretty old ffmpeg (which CLI might be different).

**Comment by Artiume** [comment]
@geilername i'm getting more reports of hevc 10bit issues from the community

**Pull request merged**